app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb

```diff
@@ -2,6 +2,17 @@ module Authentication
   module AuthnJwt
     # Mock JWTConfiguration class to use it to develop other part in the jwt authenticator
     class ConfigurationJWTGenericVendor < ConfigurationInterface
+
+      def initialize
+        @validate_input = Authentication::AuthnJwt::ValidateInput.new
+      end
+
+      def self.validate_input(authentication_parameters)
+        @validate_input.call(
+          authentication_parameters: authentication_parameters
+        )
+      end
```

Comment:
why do we even call `ConfigurationJWTGenericVendor. validate_input(authentication_parameters )` and not just call `Authentication::AuthnJwt::ValidateInput.new. call(authentication_parameters: authentication_parameters)`?

do we see a case that this will not be the class that performs the check?

Reply:
See the @nessiLahav's remark https://github.com/cyberark/conjur/pull/2154#discussion_r629112622
There's no need to instantiate it each call...

Reply:
that's not my point. I see that in the [AuthnJwt::Authenticate class](https://github.com/cyberark/conjur/blob/f345d632cedd100369ccee91a377ebf6df26693e/app/domain/authentication/authn_jwt/authenticator.rb#L16-L15), you call the `jwt_configuration` object that implements `validate_input`. do we think that the implementations will be different? i am suggesting to move the `validate_input` as it is implemented above (by calling `Authentication::AuthnJwt::ValidateInput.new`) in the `Authenticator` class instead of here.